Update httpi dependency to 3.0

brgeen opened the pull request:
## Changes
Newer versions of Dalli (Switchboard Accounts) were having an issue
when trying to connect through the socksify dependency.
[httpi 3.0 changelog](https://my.diffend.io/gems/httpi/2.5.0/3.0.0#d2h-670634)


```
Dalli::Server#connect memcache:11211 /usr/local/bundle/gems/socksify-1.7.1/lib/socksify.rb:178:in `initialize': no implicit conversion of Hash into String (TypeError)  initialize_tcp host, port, local_host, local_port
```

httpi 3.0 no longer uses the socksify dependency.


## Test
1. Point the gem to this branch 
 - Accounts
 - TCE 
 - Cart
```ruby
gem 'arux_app', git: "https://github.com/Arux-Software/arux_app_gem", branch: "geenen/update-httpi-version"
```

3. Bring containers and volumes down
4. Bundle
5. Build and bring containers back up
6. Log in as a public user 
7. Add a course to the cart
8. Checkout

brgeen committed:
update httpi dependency to 3.0

Newer versions of Dalli were having an issue
when trying to connect through the socksify
dependency.

httpi 3.0 no longer uses the socksify
dependency

chriswarren reviewed (approved):
thanks for linking to that diff/changelog

nhaglind reviewed (approved):
LGTM. Seems this is pretty minimal and doesn't depend on much itself even in later versions. https://rubygems.org/gems/httpi/versions/3.0.2

brgeen committed:
bump the semantic version

Patch fix for incompatible gem dependency
httpi when being used with new versions of Dalli.
